protocols/abci/addressing-1174.md

```diff
@@ -0,0 +1,230 @@
+# Allowing Non-Determinism in `ProcessProposal` (a.k.a. addressing #1174)
+
+## Context
+
+### Consensus Properties
+
+Byzantine Fault Tolerant (BFT) Consensus is usually specified by the following properties (2 invariants and the liveness property termination):
+
+- _agreement_: no two correct processes decide differently.
+- _validity_: the decided block `v` satisfies the predefined predicate `valid(v)`.
```

Comment:
This is a long discussion, but BFT consensus definitions do not necessarily include this property.

Tendermint consensus adopts a variant of the BFT consensus definition that relies on an _external_ validity check for proposed values.

Reply:
I took this spec from elsewhere in our own spec directory (don't remember where exactly)